### PR TITLE
Adding Activity fragment to support runtime permission check.

### PR DESCRIPTION
**Pull request by claywilkinson**
This creates an Activity Fragment that handles the flow of checking the runtime permissions.  The native code calls this with a callback which returns the result of the permission check.

### CODE REVIEW

**Comment by claywilkinson** [comment]
@proppy, @ggfan - can you take a look at this PR?

**Comment by proppy** [comment]
Does this replace #96?

**Comment by claywilkinson** [comment]
Yes, I think so.  This is a more general solution that could easily be used in other native samples.

**Comment by ggfan** [comment]
@proppy, may you have a look?  I might be look at this at the end of the week

File: audio-echo/app/build.gradle [hunk 1]
```diff
@@ -30,3 +30,12 @@ model {
         create('all')
     }
 }
+configurations {
+    testCompile
```
Comment (proppy) [inline audio-echo/app/build.gradle]: is there some tests?

File: audio-echo/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
+
     <application
         android:allowBackup="false"
         android:fullBackupContent="false"
         android:supportsRtl="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
-        android:theme="@style/AppTheme" >
+        android:theme="@style/Theme.AppCompat.Light" >
```
Comment (proppy) [inline audio-echo/app/src/main/AndroidManifest.xml]: should we do that across all sample, if yes I'll file a bug.

**Comment by proppy** [comment]
A few high level comments before diving more into this tomorrow.

- it would be nice to show how to do the fragment transaction in a c++ sample using native activity that doesn't have any Java code
- we should check that this is covering all the comments @jfschmakeit did in #96, maybe he can be ultra-generous and take the time to review this as well?

File: audio-echo/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (claywilkinson) [inline audio-echo/app/src/main/AndroidManifest.xml]: I think it is a good idea, but not critical.  There are a lot of new UI elements that require the appcompat library on older versions of Android.  I specifically changed this one for the UI elements used to display the permissions prompt and rationale.

**Comment by claywilkinson** [comment]
Updated the build.gradle file.  OK to merge @proppy?

**Comment by proppy** [comment]
@claywilkinson it would be cool if @jfschmakeit could proof read it before we merge to make sure it covers everything he reviewed as part of #96.

**Comment by hak** [comment]
Clayton, could you resolve the conflict and re-submit the patch?

**Comment by claywilkinson** [comment]
Abandoning this one and opened #189

**Pull request closed without merging**